shaman/controllers/nodes/__init__.py

```diff
@@ -42,6 +43,14 @@ def index(self):
             resp[node.host] = node.__json__()
         return resp
 
+    @secure(basic_auth)
+    @expose(method='GET', content_type="text/plain")
+    def next(self):
+        next_node = get_next_node()
+        if not next_node:
+            abort(500, "There are no healthy chacra nodes available.")
```

Comment:
this should be a `404` I think. Because it is not finding any hosts to give. This doesn't look like an internal error state